sdk/storage/azure-storage-files-shares/src/share_file_client.cpp

```diff
@@ -489,11 +484,11 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     protocolLayerOptions.ContentLength = content.Length();
     protocolLayerOptions.XMsRange = std::string("bytes=") + std::to_string(offset)
         + std::string("-") + std::to_string(offset + content.Length() - 1);
-    if (options.TransactionalContentHash.HasValue()
-        && options.TransactionalContentHash.Value().Algorithm != HashAlgorithm::Md5)
-    {
-      std::abort();
-    }
+
+    AZURE_ASSERT(
+        options.TransactionalContentHash.HasValue()
+        && options.TransactionalContentHash.Value().Algorithm == HashAlgorithm::Md5);
```

Comment:
not equivalent

Reply:
I don't think this is a contract that should be asserted.  Why is the client validating this and not sending it on to the server?  Service is the source of truth.

Reply:
Service ignores this header.

Reply:
@Jinming-Hu , I think that should be an issue for the service to ask them to return an error on that case if we think is not good. The SDK should not provide patches for Service behavior.

If the server ever change the behavior in a different way as how the SDK decided, wouldn't it be a breaking change which might be even worst for customers?

Reply:
> If the server ever change the behavior in a different way as how the SDK decided, wouldn't it be a breaking change which might be even worst for customers?

If server supports CRC64 in the future, we can just remove the assertion. That's not a breaking change.

Reply:
We can discuss this further. But don't make any changes to the behavior in this PR, to keep PR single-purpose.